### ISSUE TITLE
New color balancing and conversion pipeline

### ISSUE THREAD

**Issue by bouncyball-git** [ilia3101/MLV-App#84]
The discussion was started:

[Here](https://github.com/ilia3101/MLV-App/commit/c22488966b96a0b38fb38c66238210f3b3c98a44)
[Here](https://github.com/ilia3101/MLV-App/commit/dcaf7c70381b6c5f1389226dd2779b8850b9a458)
[Here](https://github.com/ilia3101/MLV-App/commit/0e9025ec39407d8796534ad109a5c460c60caf9f)
[Here](https://github.com/ilia3101/MLV-App/commit/6d0d21d183ce36d59ae7155194ba43b91d8ba37a)

The low Kelvin (<4000) footage is yet uncorrectable, could not be white balanced.

**Comment by bouncyball-git** [comment]
@masc4ii 

These screenshots are made with v0.17. Comparison between MLV App and Adobe Lightroom WB Pickup:
![patch1](https://user-images.githubusercontent.com/17232233/42388067-613131bc-8155-11e8-9e54-0bb4fa198ab5.jpg)
![patch2](https://user-images.githubusercontent.com/17232233/42388068-61520324-8155-11e8-99ff-9487e3330cfd.jpg)
![patch3](https://user-images.githubusercontent.com/17232233/42388069-61760b34-8155-11e8-909d-6462600b432a.jpg)
![patch4](https://user-images.githubusercontent.com/17232233/42388070-619e1e94-8155-11e8-967b-30f16037952e.jpg)
![patch5](https://user-images.githubusercontent.com/17232233/42388071-61c05b3a-8155-11e8-9cde-bee9adb2fb74.jpg)
Kelvin is quite close but tint is very off.

**Comment by bouncyball-git** [comment]
@ilia3101 

What about revising WhiteBalance branch (reworking multiplier deriving) and trying to merge it to the master? I realy loved the colors it produced in range 5000-7000K.

**Comment by ilia3101** [comment]
Sorry again about slowness of replying. I think I even saw it like in 1 day but went to do something else.

I have considered trying to put that stuff back in to master, the biggest problem is that WhiteBalance completely ruined highlight reconstruction. I would like to rewrite the work I did in white balance branch in a way that won't break highlight reconstruction.

The matrix part of the code in WhiteBalance is also a bit too spaghetti to understand I wrote so that's another reason to re write it.

I did also recently discover why colour was wrong at some temperatures in WhiteBalance: the pure blackbody curve is not used for white balance, there is a special daylight curve (or D curve, has D65 D55 points on it). I need to find a good look up table for it or calculate it somehow, would fix all the tint problems I think.
